crates/polars-core/src/chunked_array/mod.rs

```diff
@@ -181,8 +181,21 @@ impl<T: PolarsDataType> ChunkedArray<T> {
         self.bit_settings.remove(Settings::FAST_EXPLODE_LIST)
     }
 
-    pub(crate) fn clear_settings(&mut self) {
-        self.bit_settings.bits = 0;
+    pub(crate) fn get_flags(&self) -> u8 {
+        self.bit_settings.bits
+    }
+
+    /// Set flags for the Chunked Array
+    pub(crate) fn set_flags(&mut self, flags: u8) -> PolarsResult<()> {
+        Settings::from_bits(flags)
+            .ok_or_else(|| {
+                PolarsError::ComputeError(
```

Comment:
For consistency we should use the `polars_err!` macro. 

replace with `polars_err!(ComputeError: "corrupt flags {} for {}", flags, self.dtype())`